core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -280,7 +281,8 @@ class KafkaServer(
         _brokerState = BrokerState.RECOVERY
         logManager.startup(zkClient.getAllTopicsInCluster())
 
-        remoteLogManager = createRemoteLogManager(config)
+        val remoteLogManagerConfig = new RemoteLogManagerConfig(config)
```

Comment:
should we create this only when `remoteLogManagerConfig.enableRemoteStorageSystem()`?

Reply:
Good call